lib/Connection.js

```diff
@@ -16,82 +16,71 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-const iXml = require('./ixml');
-const { iRestHttp } = require('./irest');
-const { db2Call } = require('./istoredp');
+const {
+  iXmlNodeHead, iXmlNodeScriptOpen, iXmlNodeScriptClose,
+} = require('./ixml');
 
-const I_TRANSPORT_REST = 'REST';
-const I_TRANSPORT_DB2 = 'DB2';
+const { iRestHttp } = require('./transports/irest');
+const { db2Call } = require('./transports/istoredp');
+
+const availableTransports = {
+  idb: db2Call,
+  rest: iRestHttp,
+};
 
 class Connection {
   /**
    * @description Creates a new Connection object
    * @constructor
-   * @param {string} db
+   * @param {object | string} optionsOrDb
    * @param {string} user
    * @param {string} pwd
-   * @param {object} [option]
+   * @param {object} [restConfig]
    */
-  constructor(db, user, pwd, option) {
-    this.conn = {};
-    this.cmds = [];
-    this.timeout = 5000; // Default timeout is 5 seconds for sync mode.
-    this.I_DEBUG_VERBOSE = false;
-    this.conn.I_TRANSPORT_DB2_DATABASE = db; // Required Field
-    this.conn.I_TRANSPORT_DB2_USER = user; // Required Field
-    this.conn.I_TRANSPORT_DB2_PASSWORD = pwd; // Required Field
-    this.conn.I_XML_SERVICE_LIB = 'QXMLSERV'; // Default XML Service library
-    this.conn.I_TRANSPORT_CTL = '*here';
-    this.conn.I_TRANSPORT_IPC = '*NA';
-
-    if (typeof db === 'object') {
-      this.conn.I_TRANSPORT = I_TRANSPORT_DB2;
-      return;
-    }
-
-    if (option && typeof option === 'object') {
-      if (option.host) {
-        this.conn.I_TRANSPORT = I_TRANSPORT_REST;
-        this.conn.I_TRANSPORT_REST_HOST = option.host;
-        if (option.port) {
-          this.conn.I_TRANSPORT_REST_PORT = option.port;
-        } else {
-          this.conn.I_TRANSPORT_REST_PORT = 80;
-        }
-        if (option.path) {
-          this.conn.I_TRANSPORT_REST_PATH = option.path;
-        } else {
-          this.conn.I_TRANSPORT_REST_PATH = '/';
-        }
-      } else {
-        this.conn.I_TRANSPORT = I_TRANSPORT_DB2;
-        if (option.xslib && option.xslib.length > 0) {
-          this.conn.I_XML_SERVICE_LIB = option.xslib;
+  constructor(optionsOrDb, username, password, restOptions) {
+    this.commandList = [];
+    this.returnError = true;
+    this.verbose = false;
+
+    let options = optionsOrDb;
+    // maintain compatibility with pre 1.0 version
+    if (typeof options !== 'object') {
+      options = {};
+      options.database = optionsOrDb;
+      options.username = username;
+      options.password = password;
```

Comment:
are `database`, `username`, and `password` not transport options?

Reply:
Yes will add those properties to `options.transportOptions` sub object.